sql/driver/conn.go

```diff
@@ -46,56 +46,45 @@ func (c *conn) Begin() (driver.Tx, error) {
 }
 
 func (c *conn) Exec(stmt string, args []driver.Value) (driver.Result, error) {
-	rows, err := c.Query(stmt, args)
+	rows, err := c.query(stmt, args)
 	if err != nil {
 		return nil, err
 	}
 	return driver.RowsAffected(len(rows.rows)), nil
 }
 
-func (c *conn) Query(stmt string, args []driver.Value) (*rows, error) {
+func (c *conn) Query(stmt string, args []driver.Value) (driver.Rows, error) {
+	return c.query(stmt, args)
+}
+
+func (c *conn) query(stmt string, args []driver.Value) (*rows, error) {
```

Comment:
Wasn't obvious at first glance why the separation between `Query` and `query`. I might keep the names the same myself if I wrote this, but on reviewing it was a bit confusing. Perhaps `s/query/internalQuery/g` or something like that would help.